sql/moz-fx-data-marketing-prod/acoustic/contact_v1/query.sql

```diff
@@ -0,0 +1,155 @@
+      ) AS sub_mozilla_fellowship_awardee_alumni,  -- # nulls  // boolean
+      CAST(
+        COALESCE(CAST(sub_mozilla_festival AS INTEGER), 0) AS BOOLEAN
+      ) AS sub_mozilla_festival,  -- 2 nulls  // boolean
+      CAST(
+        COALESCE(CAST(sub_mozilla_technology AS INTEGER), 0) AS BOOLEAN
+      ) AS sub_mozilla_technology,  -- 2 nulls  // boolean
+      CAST(
+        COALESCE(CAST(sub_mozillians_nda AS INTEGER), 0) AS BOOLEAN
+      ) AS sub_mozillians_nda,  -- 2 nulls  // boolean
+      CAST(
+        COALESCE(CAST(sub_firefox_accounts_journey AS INTEGER), 0) AS BOOLEAN
+      ) AS sub_firefox_accounts_journey,  -- 2 nulls  // boolean
+      CAST(
+        COALESCE(CAST(sub_knowledge_is_power AS INTEGER), 0) AS BOOLEAN
+      ) AS sub_knowledge_is_power,  -- 2 nulls  // boolean
+      CAST(
+        COALESCE(CAST(sub_take_action_for_the_internet AS INTEGER), 0) AS BOOLEAN
+      ) AS sub_take_action_for_the_internet,  -- 2 nulls  // boolean
+      CAST(
+        COALESCE(CAST(sub_test_pilot AS INTEGER), 0) AS BOOLEAN
+      ) AS sub_test_pilot,  -- 2 nulls  // boolean
+      CAST(
+        COALESCE(CAST(sub_firefox_news AS INTEGER), 0) AS BOOLEAN
+      ) AS sub_firefox_news,  -- 2 nulls  // boolean
+      CAST(
+        COALESCE(CAST(sub_about_mozilla AS INTEGER), 0) AS BOOLEAN
+      ) AS sub_about_mozilla,  -- 13436 nulls  // boolean
+      CAST(
+        COALESCE(CAST(sub_apps_and_hacks AS INTEGER), 0) AS BOOLEAN
+      ) AS sub_apps_and_hacks,  -- 12988 nulls  // boolean
+      CAST(
+        COALESCE(CAST(sub_rally AS INTEGER), 0) AS BOOLEAN
+      ) AS sub_rally,  -- 13641 nulls  // boolean
+      CAST(
+        COALESCE(CAST(NULLIF(sub_firefox_sweepstakes, '') AS INTEGER), 0) AS BOOLEAN
+      ) AS sub_firefox_sweepstakes,  -- 14226 nulls  // boolean
+      LOWER(
+        NULLIF(vpn_waitlist_geo, '')
+      ) AS vpn_waitlist_geo,  -- 1221463 nulls, contains case diffences  // no_value is a bad choice of default, should this just be null?
+      SPLIT(
+        NULLIF(vpn_waitlist_platform, ''),
+        ","
+      ) AS vpn_waitlist_platform,  -- 1230937 nulls, single field contains multiple values comma separated, should this be a repeated field?  // repeated string field
+      relay_waitlist_geo,  -- # missing field, need to verify, maybe it's related to reusing old connector
+      RECIPIENT_ID AS recipient_id,  -- Acoustic system field
+      FORMAT_FXA_DATE(
+        Last_Modified_Date
+      ) AS acoustic_last_modifled_date,  -- Acoustic system field
+      @submission_date AS job_date,
+      CURRENT_DATETIME() AS processed_at,
+    FROM
+      `moz-fx-data-bq-fivetran.acoustic_sftp.contact_export`
+  )
+
+  , YESTERDAY_LOADED_CONTACTS AS (
+      SELECT * EXCEPT(job_date, processed_at)
+      FROM `moz-fx-data-marketing-prod.acoustic.contact_v1`  -- possible that this may not exist on first run
+      WHERE job_date = DATE_SUB(@submission_date, INTERVAL 1 DAY)
+  )
+
+
+  SELECT * FROM JUST_LOADED_CONTACTS
```

Comment:
@ANich what do you think about an approach like this? so that over time we just collect the "current" state of all customers as of today. This should enable us to do nice aggregation of number of subscribers to each product over time?

Reply:
Is the final table supposed to be one row per contact per day? I'm not clear on how the processing/job date affects the granularity/partitioning. My intuition is that it shouldn't unless we have to but I don't have all the context. -- Will it be possible to recreate the table from scratch if necessary with this approach?

Reply:
okay, as per our discussion I updated the logic to just capture changes as we get them from Acoustic. We can always create another process on top of this "changelog" to generate the current version of contacts dataset.